docs/flows/flows.puml

```diff
@@ -46,83 +47,155 @@ autonumber "Current:[00]"
-community <-- forum: proposals
+alice -> forum: create forum proposal
+activate forum
+community -> forum: get proposals
+community <-- forum: list of proposals
 community -> forum: put positive/negative feedback
-
 alice <-- forum: get positive/negative feedback
+deactivate forum
 
 community -> community: delegate votes
 
-alice -> snapshot: if (positive) submit proposal
-
-tokenholders -> snapshot: show proposals
-tokenholders <-- snapshot: proposals
+alice -> snapshot: if (positive feedback) submit proposal
+activate snapshot
+tokenholders -> snapshot: get proposals
+tokenholders <-- snapshot: list of proposals
 tokenholders -> snapshot: vote on open proposals
-tokenholders <- snapshot: vote
-bitdaoadmin -> bitdaotreasury: if (pass) allow funds
-bitdaotreasury -> grantstreasury: transfer funds
+tokenholders <-- snapshot: vote confirmation
+
+'alice -> snapshot: get Alice's proposal result
+'alice <-- snapshot: Alice's proposal pass/fail result
+
+bitdaoadmin -> snapshot: get Alice's proposal result
+bitdaoadmin <-- snapshot: Alice's proposal pass/fail result
+deactivate snapshot
+
+bitdaoadmin -> bitdaotreasury: if (proposal passed) transfer funds to grants treasury
+bitdaotreasury -> grantstreasury: funds transfer
 
 alt#Gold #DarkSeaGreen trace
-  alice <-- grantstreasury: TxHash
+  alice <-- grantstreasury: transfer confirmation
 end
 
-alice <-- grantstreasury: execute proposal
+alice --> alice: execute proposal
 
 newpage
 
 autonumber "Bond:[00]"
 
 == Bond ==
 
+
+alice -> grantstreasury: create grants treasury
+alice <-- grantstreasury: treasury address
+
+' bond
+
 alice -> forum: create forum proposal
-alice <- guarantor: positive/negative feedback
+activate forum
+guarantor -> forum: get proposals
+guarantor <-- forum: list of proposals
+guarantor -> forum: put positive/negative feedback
+
 alice -> bondcontract: if (positive) create bond
+guarantor -> bondcontract: if (positive) contribute BIT tokens
+guarantor <-- bondcontract: ERC20 Fungible bond tokens (1:1 swap)
 
-guarantor -> bondcontract: contribute BIT tokens
-bondcontract <-- guarantor: ERC20 Fungible bond tokens (1:1 swap)
+' treasury
 
-community -> community: delegate votes
+community -> forum: get proposals
+community <-- forum: list of proposals
+community -> forum: put positive/negative feedback
+alice <-- forum: get positive/negative feedback
+deactivate forum
 
-alice -> snapshot: if (positive) submit proposal
+community --> community: delegate votes
 
-tokenholders -> snapshot: show proposals
-tokenholders <-- snapshot: proposals
+alice -> snapshot: if (positive feedback) submit proposal
+activate snapshot
+tokenholders -> snapshot: get proposals
+tokenholders <-- snapshot: list of proposals
 tokenholders -> snapshot: vote on open proposals
-tokenholders <- snapshot: vote
+tokenholders <-- snapshot: vote confirmation
+deactivate snapshot
 
-bitdaoadmin -> bitdaotreasury: if (pass) allow funds
+bitdaoadmin -> bitdaotreasury: if (pass) allow funds transfer
```

Comment:
We agreed to remove if statements as Chris suggested (if 'if' statement fails in each case assume that it simply ends flow if not specified otherwise).